virtualmachine/vsphere/vm.go

```diff
@@ -735,7 +735,12 @@ func getNicInfo(vmMo mo.VirtualMachine) []VirtualEthernetCard {
 			nwcard := c.GetVirtualEthernetCard()
 			nic.MacAddress = nwcard.MacAddress
 			nic.NicName = nwcard.DeviceInfo.GetDescription().Label
-			nic.NetworkName = nwcard.Backing.(*types.VirtualEthernetCardNetworkBackingInfo).DeviceName
+			switch b := nwcard.Backing.(type) {
+			case *types.VirtualEthernetCardNetworkBackingInfo:
+				nic.NetworkName = b.DeviceName
+			default:
+				continue
```

Comment:
So this won't return the networks attached to the NICs if they are DistributedPortGroups. The user will have to explicitly choose the networks for these NICs. The user will be able to choose DPGs for these, and they will get properly plumbed. @avnish30jn @hemanshuasolia Please confirm if this understanding is correct.

Reply:
Yes DPGs and Opaque networks won't be shown for the attached NICs, User will have to explicitly.